src/utils.rs

```diff
@@ -158,40 +160,44 @@ pub(crate) fn random_plusminus_by_size_with_minimum<R: RngCore + CryptoRng>(
 /// [`Transcript`].
 pub(crate) fn plusminus_bn_random_from_transcript(
     transcript: &mut Transcript,
-    n: &BigNumber,
-) -> BigNumber {
+    _n: &BigNumber,
+) -> Result<BigNumber> {
     let mut is_neg_byte = vec![0u8; 1];
     transcript.challenge_bytes(b"sampling negation bit", is_neg_byte.as_mut_slice());
     let is_neg: bool = is_neg_byte[0] & 1 == 1;
 
     // The sampling method samples from the open interval, so add 1 to sample from
     // the _closed_ interval we want here.
-    let open_interval_max = n + 1;
-    let b = positive_bn_random_from_transcript(transcript, &open_interval_max);
+    let q = k256_order();
+    let open_interval_max = &q + 1;
+    let b = positive_bn_random_from_transcript(transcript, &open_interval_max)?;
     match is_neg {
-        true => -b,
-        false => b,
+        true => Ok(-b),
+        false => Ok(b),
     }
```

Comment:
note: This is a perfectly good way to do this but FYI, if you have something like this with more complicated branches, you can also wrap the `Ok` around the entire match, like
```rust
Ok(match is_neg {
    true => -b,
    false => b,
})
```
In this case I don't think one is particularly better than the other, so just pick the one you prefer.